src/sema/file.rs

```diff
@@ -48,6 +48,33 @@ impl File {
         }
     }
 
+    /// Give a position as a human readable position, with only the filename not the full path.
+    pub fn loc_to_filename_string(&self, start: usize, end: usize) -> String {
+        let (from_line, from_column) = self.offset_to_line_column(start);
+        let (to_line, to_column) = self.offset_to_line_column(end);
```

Comment:
This function is a copy & paste of `fn loc_to_string()` with modifications. Please factor out the common code in a single function, e.g. pass in the filename/path as `&str` and then render the full path or filename, then call the common function.

Also use `self.path.file_name().unwrap().display()` rather than `self.path.file_name().unwrap().to_str().unwrap()`; paths are not necessarily valid utf8 on unix (although we don't have tests for this).

Reply:
file_name.unrwap returns an OsStr, which should be converted to an str first. 
I think a good solution here is:
get `fullpath: String = path.display()` first, and the parse it by hand. (this would deal with returning a valid utf8, since display is implented)